src/sys/expressionparser.cxx

```diff
@@ -149,6 +250,41 @@ FieldGeneratorPtr ExpressionParser::parseIdentifierExpr(LexInfo& lex) const {
   string name = lowercase(lex.curident);
   lex.nextToken();
 
+  // sum(symbol, count, expr)
+  // e.g. sum(i, 10, {i}^2) -> 0 + 1^2 + 2^2 + 3^2 + ... + 9^2 
+  if (name == "sum") {
```

Comment:
Just curious as to why `sum` needs hardcoding in the parser? I assume there's a reason, but it's not immediately obvious to me!

Reply:
This is because arguments to functions are currently only `FieldGenerator` objects, rather than strings. This could be done, but would probably be more complex than just hardcoding this `sum` special case. Unfortunately BOUT.inp is not a particularly elegant language...